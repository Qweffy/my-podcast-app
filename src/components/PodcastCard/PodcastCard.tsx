import { PodcastCardProps } from './PodcastCard.types.ts'

export const PodcastCard = ({ title, author, imageURL }: PodcastCardProps) => {
    return (
        <div className="flex flex-col items-center relative min-w-[5rem] max-w-[13rem]">
            <img
                src={imageURL}
                alt={title || 'Podcast Image'}
                className="absolute top-5 left-auto w-[7rem] rounded-full shadow-md aspect-square object-cover"
            />
            <div className="w-full flex flex-col gap-2 rounded-md border shadow-md mt-[5rem] pt-[4rem] min-h-[9rem] px-[.5rem] pb-[.7rem]  hover:shadow-2xl ease-in-out duration-300 ">
                <h2 className="font-semibold text-center">{title.toUpperCase()}</h2>
                <p className="text-gray-400 text-sm text-center font-semibold">Author: {author}</p>
            </div>
        </div>
    )
}
