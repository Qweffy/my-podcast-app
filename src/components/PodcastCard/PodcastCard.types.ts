export interface PodcastCardProps {
    id: string
    title: string
    author: string
    imageURL: string
}
