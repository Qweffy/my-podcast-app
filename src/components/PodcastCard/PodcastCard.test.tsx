import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { describe, it, expect } from 'vitest'
import PodcastCard from 'components/PodcastCard/index.ts'

describe('PodcastCard', () => {
    it('should render correctly with valid title, author, and imageURL', () => {
        const { container, getByText } = render(
            <PodcastCard id={'1'} title="Test Podcast" author="Test Author" imageURL="test.jpg" />,
        )
        expect(container).toMatchSnapshot()
        expect(getByText('TEST PODCAST')).toBeInTheDocument()
        expect(getByText('Author: Test Author')).toBeInTheDocument()
    })

    it('should handle empty title gracefully', () => {
        const { getByText, getByAltText } = render(
            <PodcastCard id={'2'} title="" author="Test Author" imageURL="test.jpg" />,
        )
        expect(getByAltText('Podcast Image')).toBeInTheDocument()
        expect(getByText('Author: Test Author')).toBeInTheDocument()
    })
})
