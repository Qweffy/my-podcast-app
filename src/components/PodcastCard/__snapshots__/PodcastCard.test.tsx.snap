// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PodcastCard > should render correctly with valid title, author, and imageURL 1`] = `
<div>
  <div
    class="flex flex-col items-center relative min-w-[5rem] max-w-[13rem]"
  >
    <img
      alt="Test Podcast"
      class="absolute top-5 left-auto w-[7rem] rounded-full shadow-md aspect-square object-cover"
      src="test.jpg"
    />
    <div
      class="w-full flex flex-col gap-2 rounded-md border shadow-md mt-[5rem] pt-[4rem] min-h-[9rem] px-[.5rem] pb-[.7rem]  hover:shadow-2xl ease-in-out duration-300 "
    >
      <h2
        class="font-semibold text-center"
      >
        TEST PODCAST
      </h2>
      <p
        class="text-gray-400 text-sm text-center font-semibold"
      >
        Author: 
        Test Author
      </p>
    </div>
  </div>
</div>
`;
